### PR TITLE
UCT: rdmacm on iface: null check for the UNREACHABLE event's priv_data

### PR DESCRIPTION
**Pull request by alinask**

### CODE REVIEW

File: src/uct/ib/rdmacm/rdmacm_iface.c [hunk 1]
```diff
@@ -321,11 +321,10 @@ uct_rdmacm_iface_process_event(uct_rdmacm_iface_t *iface,
                                                  sizeof(uct_rdmacm_priv_data_hdr_t));
 
             uct_rdmacm_cm_id_to_dev_name(ep->cm_id_ctx->cm_id, dev_name);
+            hdr = (uct_rdmacm_priv_data_hdr_t*)conn_param.private_data;
             /* TODO check the ep's cb_flags to determine when to invoke this callback.
              * currently only UCT_CB_FLAG_ASYNC is supported so the cb is invoked from here */
-            priv_data_ret = ep->pack_cb(ep->pack_cb_arg, dev_name,
-                                        (void*)(conn_param.private_data +
-                                        sizeof(uct_rdmacm_priv_data_hdr_t)));
+            priv_data_ret = ep->pack_cb(ep->pack_cb_arg, dev_name, hdr +1);
```
Comment (yosefe) [inline src/uct/ib/rdmacm/rdmacm_iface.c]: space after +

File: src/uct/ib/rdmacm/rdmacm_iface.c [hunk 2]
```diff
@@ -336,14 +335,12 @@ uct_rdmacm_iface_process_event(uct_rdmacm_iface_t *iface,
                 break;
             }
 
-            hdr.length = (uint8_t)priv_data_ret;
-            hdr.status = UCS_OK;
-            UCS_STATIC_ASSERT(sizeof(hdr) == sizeof(uct_rdmacm_priv_data_hdr_t));
+            hdr->length = (uint8_t)priv_data_ret;
+            hdr->status = UCS_OK;
+            UCS_STATIC_ASSERT(sizeof(*hdr) == sizeof(uct_rdmacm_priv_data_hdr_t));
```
Comment (yosefe) [inline src/uct/ib/rdmacm/rdmacm_iface.c]: this assert seems redundant now we are doing sizeof(*hdr) everywhere

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/12790/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI: <b>FAILED</b> on 9 of 29 workers (click for details)</summary>

**Note:** the logs will be deleted after 31-Oct-2019

Agent/Stage | Status
-- | --
[_main](https://gist.githubusercontent.com/mellanox-github/43d24f806db765e9d2ff1b8e38feee6a/raw/d894988ed31549aed399d330cf8eef78d4de5862/_main) | :x: FAILURE
[hpc-arm-hwi-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/43d24f806db765e9d2ff1b8e38feee6a/raw/67a0da75bd88bc55eeb05014ecddb5b9472fd62b/hpc-arm-hwi-jenkins_W0) | :x: FAILURE
[hpc-arm-hwi-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/43d24f806db765e9d2ff1b8e38feee6a/raw/6cc715a1ef9e9ba54df97eefc1a377ff19351819/hpc-arm-hwi-jenkins_W1) | :x: FAILURE
[hpc-arm-hwi-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/43d24f806db765e9d2ff1b8e38feee6a/raw/b45677ff027fe3764b19c04a06b00078f69f610b/hpc-arm-hwi-jenkins_W2) | :x: FAILURE
[hpc-test-node-legacy_W0](https://gist.githubusercontent.com/mellanox-github/43d24f806db765e9d2ff1b8e38feee6a/raw/0cc44c79974e5ccf5e0bfcad68bafa24a5a77540/hpc-test-node-legacy_W0) | :x: FAILURE
[hpc-test-node-legacy_W1](https://gist.githubusercontent.com/mellanox-github/43d24f806db765e9d2ff1b8e38feee6a/raw/8ef10cd09b1c224c3caf22ab0db6c5015ffcf5ad/hpc-test-node-legacy_W1) | :x: FAILURE
[hpc-test-node-legacy_W2](https://gist.githubusercontent.com/mellanox-github/43d24f806db765e9d2ff1b8e38feee6a/raw/435d5d8e5b10067cb54f7a250a49eee8d68399f8/hpc-test-node-legacy_W2) | :x: FAILURE
[hpc-test-node-legacy_W3](https://gist.githubusercontent.com/mellanox-github/43d24f806db765e9d2ff1b8e38feee6a/raw/59dd9ee41cd6360e3b2698ab47b8c4c1f735732f/hpc-test-node-legacy_W3) | :x: FAILURE
[hpc-test-node-new_W1](https://gist.githubusercontent.com/mellanox-github/43d24f806db765e9d2ff1b8e38feee6a/raw/53f6e743435423808dadcf58479ddab1d9399b17/hpc-test-node-new_W1) | :x: FAILURE
[hpc-arm-cavium-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/43d24f806db765e9d2ff1b8e38feee6a/raw/ab11522e4ddc350095c3a04905de893c09d5e0f4/hpc-arm-cavium-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/43d24f806db765e9d2ff1b8e38feee6a/raw/712731b011dc6cc172e0d68cb2464ca048dffd6a/hpc-arm-cavium-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/43d24f806db765e9d2ff1b8e38feee6a/raw/97e464aefe7c3896163b4de52ad471c4599a856c/hpc-arm-cavium-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/43d24f806db765e9d2ff1b8e38feee6a/raw/7fe364776778b2771dafcc8835b4c3545db0da51/hpc-arm-cavium-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/43d24f806db765e9d2ff1b8e38feee6a/raw/02ebee54b5f9aaddfd6584e6f33ee88314a77215/hpc-arm-hwi-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W0](https://gist.githubusercontent.com/mellanox-github/43d24f806db765e9d2ff1b8e38feee6a/raw/80c4254fde1bdd015376988575562526878d859c/hpc-test-althca_W0) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W1](https://gist.githubusercontent.com/mellanox-github/43d24f806db765e9d2ff1b8e38feee6a/raw/218ff68e1bf57f7440fe9596bfc6a695f0ad99ce/hpc-test-althca_W1) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W2](https://gist.githubusercontent.com/mellanox-github/43d24f806db765e9d2ff1b8e38feee6a/raw/6f3af4b9221df0d29f943a96d32ed8dd1e4040a3/hpc-test-althca_W2) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W3](https://gist.githubusercontent.com/mellanox-github/43d24f806db765e9d2ff1b8e38feee6a/raw/78af9fe46a71ecb82b5599e5aa90ccd39c846c10/hpc-test-althca_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W0](https://gist.githubusercontent.com/mellanox-github/43d24f806db765e9d2ff1b8e38feee6a/raw/657bff7b15c83c06690901205ef5d226fc3caa2e/hpc-test-node-gpu_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W1](https://gist.githubusercontent.com/mellanox-github/43d24f806db765e9d2ff1b8e38feee6a/raw/d5adcd827cc7465c57c8e838f0fe74322e2734e2/hpc-test-node-gpu_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W2](https://gist.githubusercontent.com/mellanox-github/43d24f806db765e9d2ff1b8e38feee6a/raw/07202986b55bf38d78b1c6b8f227d8618ec76d1a/hpc-test-node-gpu_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W3](https://gist.githubusercontent.com/mellanox-github/43d24f806db765e9d2ff1b8e38feee6a/raw/29de333fa1cdc27601d97b449aa07ff133f711e4/hpc-test-node-gpu_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W0](https://gist.githubusercontent.com/mellanox-github/43d24f806db765e9d2ff1b8e38feee6a/raw/893d40af38054c2c9df2bccd5d6aaf4e53c59633/hpc-test-node-new_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W2](https://gist.githubusercontent.com/mellanox-github/43d24f806db765e9d2ff1b8e38feee6a/raw/b5550dff94711de842cba697965056b41bf2e3e1/hpc-test-node-new_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W3](https://gist.githubusercontent.com/mellanox-github/43d24f806db765e9d2ff1b8e38feee6a/raw/02395ad4a22a9547604b793949eb466f1b3749d8/hpc-test-node-new_W3) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/43d24f806db765e9d2ff1b8e38feee6a/raw/dafba8223e693a2ebde90ea508f857374a8e5dd3/r-vmb-ppc-jenkins_W0) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/43d24f806db765e9d2ff1b8e38feee6a/raw/6a8cfb78ffa24acab4757c329feb9cbb16a57ebf/r-vmb-ppc-jenkins_W1) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/43d24f806db765e9d2ff1b8e38feee6a/raw/d27123ae09af5c88de5a51be1b7853ddbd020b62/r-vmb-ppc-jenkins_W2) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/43d24f806db765e9d2ff1b8e38feee6a/raw/b1c5f199ccdbf32988cc5c69ece4aeccfe643d49/r-vmb-ppc-jenkins_W3) | :heavy_check_mark: SUCCESS

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/12792/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI: <b>PASSED</b#  > on 29 workers (click for details)</summary>

**Note:** the logs will be deleted after 31-Oct-2019

Agent/Stage | Status
-- | --
[_main](https://gist.githubusercontent.com/mellanox-github/b41fd6f9ee10fc974de97d45ecfa2f14/raw/98885d50bf4a0ae58526e1705638d8263290284c/_main) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/b41fd6f9ee10fc974de97d45ecfa2f14/raw/9f9443e713b7e9eff49f52d1f4ec5b6b7fe29ed6/hpc-arm-cavium-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/b41fd6f9ee10fc974de97d45ecfa2f14/raw/a4546e67a9801b6abc859044ca1c70b15c24c24f/hpc-arm-cavium-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/b41fd6f9ee10fc974de97d45ecfa2f14/raw/bc7164ac34de6bd15b529799264d971c78f8734a/hpc-arm-cavium-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/b41fd6f9ee10fc974de97d45ecfa2f14/raw/0ec8dbb6959c9f563fb70a183e574a36a1a1fb48/hpc-arm-cavium-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/b41fd6f9ee10fc974de97d45ecfa2f14/raw/b345bf0f03109bb608b121c3d57750fa83bcdb55/hpc-arm-hwi-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/b41fd6f9ee10fc974de97d45ecfa2f14/raw/d523d63e5f3bccf7376eeba80d42980a5b0af00b/hpc-arm-hwi-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/b41fd6f9ee10fc974de97d45ecfa2f14/raw/9cac4a5754296ef7f2e077b94f713a5e5510e9e6/hpc-arm-hwi-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/b41fd6f9ee10fc974de97d45ecfa2f14/raw/8cda14c0c07f1430787731c0cb27cd98eb7e866f/hpc-arm-hwi-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W0](https://gist.githubusercontent.com/mellanox-github/b41fd6f9ee10fc974de97d45ecfa2f14/raw/b4733c1bbcde79e0152c0bdfb2acc6ba50eb0d06/hpc-test-althca_W0) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W1](https://gist.githubusercontent.com/mellanox-github/b41fd6f9ee10fc974de97d45ecfa2f14/raw/dcea4053946c091a9a3e9a918e471aa819c4cf9a/hpc-test-althca_W1) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W2](https://gist.githubusercontent.com/mellanox-github/b41fd6f9ee10fc974de97d45ecfa2f14/raw/7e20da6f21b7e9a7ad056afaa3994b63fd569484/hpc-test-althca_W2) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W3](https://gist.githubusercontent.com/mellanox-github/b41fd6f9ee10fc974de97d45ecfa2f14/raw/4e66b47a26ebe64e0346c4570f0212cd7b6605ed/hpc-test-althca_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W0](https://gist.githubusercontent.com/mellanox-github/b41fd6f9ee10fc974de97d45ecfa2f14/raw/180ded81260666896f24f9ed8fa18820631fd596/hpc-test-node-gpu_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W1](https://gist.githubusercontent.com/mellanox-github/b41fd6f9ee10fc974de97d45ecfa2f14/raw/d5a390aa51b2fb5628dd00808f692a8ec0b42aba/hpc-test-node-gpu_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W2](https://gist.githubusercontent.com/mellanox-github/b41fd6f9ee10fc974de97d45ecfa2f14/raw/4463de80239e0a71f831c30b891b59f86a2a5b0f/hpc-test-node-gpu_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W3](https://gist.githubusercontent.com/mellanox-github/b41fd6f9ee10fc974de97d45ecfa2f14/raw/c03d9a3316dfd784463ce9a09bebc8926f0290b0/hpc-test-node-gpu_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W0](https://gist.githubusercontent.com/mellanox-github/b41fd6f9ee10fc974de97d45ecfa2f14/raw/61a1ed453e6e0b41af8a06f1a4bff879892aa8d0/hpc-test-node-legacy_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W1](https://gist.githubusercontent.com/mellanox-github/b41fd6f9ee10fc974de97d45ecfa2f14/raw/2b33db2c83483970adbb617cc078f61b455a5654/hpc-test-node-legacy_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W2](https://gist.githubusercontent.com/mellanox-github/b41fd6f9ee10fc974de97d45ecfa2f14/raw/9c5e62fa4fa261e87c5ea9b80827f852066e9100/hpc-test-node-legacy_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W3](https://gist.githubusercontent.com/mellanox-github/b41fd6f9ee10fc974de97d45ecfa2f14/raw/f911a83df4c10cf5132e5d256559e4d51613d678/hpc-test-node-legacy_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W0](https://gist.githubusercontent.com/mellanox-github/b41fd6f9ee10fc974de97d45ecfa2f14/raw/113eac0aaae8c015948971ce7241d24221eaae9d/hpc-test-node-new_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W1](https://gist.githubusercontent.com/mellanox-github/b41fd6f9ee10fc974de97d45ecfa2f14/raw/1f0ad3c728fb70053c7e91e8a87b20dbe370f4ab/hpc-test-node-new_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W2](https://gist.githubusercontent.com/mellanox-github/b41fd6f9ee10fc974de97d45ecfa2f14/raw/96338bc0cf5cefde9ef3264d5ebb7ae120e4c287/hpc-test-node-new_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W3](https://gist.githubusercontent.com/mellanox-github/b41fd6f9ee10fc974de97d45ecfa2f14/raw/bc93ffed40402ec0c5708d8d9eac1c380d3c06fe/hpc-test-node-new_W3) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/b41fd6f9ee10fc974de97d45ecfa2f14/raw/ca96246d7849d40c4c9290c20faf024093c04991/r-vmb-ppc-jenkins_W0) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/b41fd6f9ee10fc974de97d45ecfa2f14/raw/1cd2b6bd8344ab2f9b5cc150a5b4737574e0c898/r-vmb-ppc-jenkins_W1) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/b41fd6f9ee10fc974de97d45ecfa2f14/raw/0c0d3649282de3c21dccb9cb03ab1649d4b52026/r-vmb-ppc-jenkins_W2) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/b41fd6f9ee10fc974de97d45ecfa2f14/raw/d0d8541716c7fc4343a500d9d638e3e36441586f/r-vmb-ppc-jenkins_W3) | :heavy_check_mark: SUCCESS

**Comment by yosefe** [comment]
bot:pipe:retest

**Comment by alinask** [comment]
@yosefe good to go?

**Pull request merged**